boards/BMS/Src/App/states/App_ChargeState.c

```diff
@@ -44,23 +52,31 @@ static void ChargeStateRunOnTick100Hz(struct StateMachine *const state_machine)
             has_charger_faulted = App_Charger_HasFaulted(charger);
         }
 
-        uint8_t    segment = 0U;
-        uint8_t    cell    = 0U;
-        const bool has_reached_max_v =
-            App_Accumulator_GetMaxVoltage(accumulator, &segment, &cell) > MAX_CELL_VOLTAGE_THRESHOLD;
+        if (is_charger_disconnected || has_charger_faulted)
+        {
+            next_state = App_GetFaultState();
+        }
+        else if (!is_pack_balanced && App_Accumulator_IsBalancingAllowed(accumulator))
+        {
+            // Start balancing when the pack is unbalanced, and that the minimum cell voltage is
+            next_state = App_GetBalanceState();
+        }
+        else if (is_pack_balanced && has_reached_max_v)
```

Comment:
Again, you designed a function in App_Accumulator to check this

Reply:
> thought we needed to use `has_reached_max_v` but realized we dont rlly need it. should be improved now

^ addressed here